text_analyzer.py

```diff
@@ -0,0 +1,162 @@
+    garpike and stingray are also present."""
+]
+SEPARATOR = 42 * "-"
+USERS = dict(bob="123", ann="pass123", mike="password123", liz="pass123")
+
+
+# Pomocná datová struktura pro jednoduší manipulaci
+class NumberStats:
+    def __init__(self):
+        self.titlecase = 0
+        self.uppercase = 0
+        self.lowercase = 0
+        self.numeric = 0
+        self.summed = 0
+        self.words = 0
+
+
+# Třída s metodami pro jednoduší manipulaci
+class Stats:
+    def __init__(self):
+        self.numbers = NumberStats()
+        self.histogram = dict()
+        self.sorted_histogram = list()
+
+    def print_histogram(self):
+        print("Histogram délky slov:")
+        for item in self.sorted_histogram:
+            print(item, "*" * self.histogram[item], self.histogram[item])
+
+    def print_stats(self):
+        print(f"Ve zvoleném textu je celkem {self.numbers.words} slov:",
+              f"    {self.numbers.titlecase}krát slovo začínající velkým písmenem",
+              f"    {self.numbers.uppercase}krát slovo psané velkými písmeny",
+              f"    {self.numbers.lowercase}krát slovo psané malými písmeny",
+              f"    {self.numbers.numeric}krát číselný řetězec",
+              f"Celkový součet hodnot číselných řetězců je {self.numbers.summed} ",
+              sep='\n')
+
+    def fill_stats(self, text_string):
+        string_to_analyze = text_string.split()
+
+        for word in string_to_analyze:
+
+            word_clean = word.strip(',.')
+
+            if word_clean.istitle():
+                self.numbers.titlecase += 1
+
+            elif word_clean.isupper():
+                self.numbers.uppercase += 1
+
+            elif word_clean.islower():
+                self.numbers.lowercase += 1
+
+            elif word_clean.isnumeric():
+                self.numbers.numeric += 1
+
+                self.numbers.summed += int(word_clean)
+
+            self.histogram[len(word_clean)] = self.histogram.get(len(word_clean), 0) + 1
+
+        # Celkový počet slov v textu
+        self.numbers.words = len(string_to_analyze)
+
+        # Seřazený histogram - klíče od nejmenšího po největší
+        self.sorted_histogram = sorted(self.histogram)
+
+
+def login(users_db):
+    while True:
+        username = input("Uživatelské jméno:").strip()
+        password = input("Heslo:")
+
+        # Test hesla
+        if username in users_db.keys() and password == users_db.get(username, 0):
+            # Heslo je správné -> vyskočit ze smyčky
+            break
+
+        else:
+            print(f"Uživatel {username} nemá přístup nebo zadal špatné heslo")
+            print("Zadejte údaje znovu")
+            # Heslo je nesprávné -> pokračovat ve smyčce
+            continue
+
+
+def choose_text(text_string):
+    # Počet dostupných textů
+    text_count = len(text_string)
+    print(f"Zvolte jeden z {text_count} textů k analýze")
+
+    while True:
+
+        selection = int(input(f"Zvolte číslo textu (1-{text_count}):"))
```

Comment:
If user enters a non-number (e.g. `a`) this will fail. You should warn her about that.